src/main/rx/flysky.c

```diff
@@ -215,167 +198,126 @@ static bool isValidPacket (const uint8_t *packet) {
     return (rcPacket->rxId == rxId && rcPacket->txId == txId);
 }
 
-static void buildAndWriteTelemetry (uint8_t *packet)
-{
+static void buildAndWriteTelemetry (uint8_t *packet) {
     if (packet) {
         static uint8_t bytesToWrite = FLYSKY_2A_PAYLOAD_SIZE; // first time write full packet to buffer a7105
         flySky2ATelemetryPkt_t *telemertyPacket = (flySky2ATelemetryPkt_t*) packet;
         uint16_t voltage = 10 * getBatteryVoltage();
-
         telemertyPacket->type = FLYSKY_2A_PACKET_TELEMETRY;
-
         telemertyPacket->sens[0].type = SENSOR_INT_V;
         telemertyPacket->sens[0].number = 0;
         telemertyPacket->sens[0].valueL = voltage & 0xFF;
         telemertyPacket->sens[0].valueH = (voltage >> 8) & 0xFF;
-
         telemertyPacket->sens[1].type = SENSOR_RSSI;
         telemertyPacket->sens[1].number = 0;
         telemertyPacket->sens[1].valueL = rssi_dBm & 0xFF;
         telemertyPacket->sens[1].valueH = (rssi_dBm >> 8) & 0xFF;
-
         telemertyPacket->sens[2].type = SENSOR_ERR_RATE;
         telemertyPacket->sens[2].number = 0;
         telemertyPacket->sens[2].valueL = errorRate & 0xFF;
         telemertyPacket->sens[2].valueH = (errorRate >> 8) & 0xFF;
-
         memset (&telemertyPacket->sens[3], 0xFF, 4 * sizeof(flySky2ASens_t));
-
         A7105WriteFIFO(packet, bytesToWrite);
-
         bytesToWrite = 9 + 3 * sizeof(flySky2ASens_t);// next time write only bytes that could change, the others are already set as 0xFF in buffer a7105
     }
 }
 
-static rx_spi_received_e flySky2AReadAndProcess (uint8_t *payload, const uint32_t timeStamp)
-{
+static rx_spi_received_e flySky2AReadAndProcess (uint8_t *payload, const uint32_t timeStamp) {
     rx_spi_received_e result = RX_SPI_RECEIVED_NONE;
     uint8_t packet[FLYSKY_2A_PAYLOAD_SIZE];
-
-    uint8_t bytesToRead = (bound) ? (9 + 2*FLYSKY_2A_CHANNEL_COUNT) : (11 + FLYSKY_FREQUENCY_COUNT);
+    uint8_t bytesToRead = (bound) ? (9 + 2 * FLYSKY_2A_CHANNEL_COUNT) : (11 + FLYSKY_FREQUENCY_COUNT);
     A7105ReadFIFO(packet, bytesToRead);
-
     switch (packet[0]) {
-        case FLYSKY_2A_PACKET_RC_DATA:
-        case FLYSKY_2A_PACKET_FS_SETTINGS: // failsafe settings
-        case FLYSKY_2A_PACKET_SETTINGS: // receiver settings
+    case FLYSKY_2A_PACKET_RC_DATA:
+    case FLYSKY_2A_PACKET_FS_SETTINGS: // failsafe settings
+    case FLYSKY_2A_PACKET_SETTINGS: // receiver settings
         if (isValidPacket(packet)) {
             checkRSSI();
             resetTimeout(timeStamp);
-
             const flySky2ARcDataPkt_t *rcPacket = (const flySky2ARcDataPkt_t*) packet;
-
             if (rcPacket->type == FLYSKY_2A_PACKET_RC_DATA) {
                 if (payload) {
-                    memcpy(payload, rcPacket->data, 2*FLYSKY_2A_CHANNEL_COUNT);
+                    memcpy(payload, rcPacket->data, 2 * FLYSKY_2A_CHANNEL_COUNT);
```

Comment:
Codacy found an issue: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6326343)